src/js/tracks/video-track-list.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file video-track-list.js
+ */
+import TrackList from './track-list';
+
+/**
+* A list of possiblee video tracks. Most functionality is in the
+ * base class Tracklist and the spec for VideoTrackList is located at:
+ * @link https://html.spec.whatwg.org/multipage/embedded-content.html#videotracklist
+ *
+ * interface VideoTrackList : EventTarget {
+ *   readonly attribute unsigned long length;
+ *   getter VideoTrack (unsigned long index);
+ *   VideoTrack? getTrackById(DOMString id);
+ *   readonly attribute long selectedIndex;
+ *
+ *   attribute EventHandler onchange;
+ *   attribute EventHandler onaddtrack;
+ *   attribute EventHandler onremovetrack;
+ * };
+ *
+ * @param {VideoTrack[]} tracks a list of video tracks to instantiate the list with
+ # @extends TrackList
+ * @class VideoTrackList
+ */
+class VideoTrackList extends TrackList {
+  constructor(tracks) {
+    let videoTrackList = super(tracks);
+    Object.defineProperty(videoTrackList, 'selectedIndex', {
+      get() {
+        for (let i = 0; i < this.length; i++) {
+          if (this[i].selected()) {
+            return i;
+          }
+        }
+        return -1;
+      },
+      set() {},
+    });
+    return videoTrackList;
+  }
+  addTrack_(track) {
+    super.addTrack_(track);
+    track.addEventListener('selectedchange', () => {
```

Comment:
in the html5 tech we wrap native audio and video tracks with our own class (still up for debate wether we should be doing that or not) so they will have an even listener in that case.